x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/search_ui_logic.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+interface InitialFieldValues {
+  validFields: string[];
+  validSortFields: string[];
+  validFacetFields: string[];
+}
+interface SearchUIActions {
+  initializeData(): void;
+  dataInitialized(initialFieldValues: InitialFieldValues): InitialFieldValues;
+  titleFieldChanged(titleField: string): { titleField: string };
+  URLFieldChanged(urlField: string): { urlField: string };
+  facetFieldsChanged(facetFields: string[]): { facetFields: string[] };
+  sortFieldsChanged(sortFields: string[]): { sortFields: string[] };
+  activeFieldChanged(activeField: string): { activeField: string };
+}
+
+interface SearchUIValues {
+  dataLoading: boolean;
+  validFields: string[];
+  validSortFields: string[];
+  validFacetFields: string[];
+  titleField: string;
+  urlField: string;
+  facetFields: string[];
+  sortFields: string[];
+  activeField: string;
+}
+
+export const SearchUILogic = kea<MakeLogicType<SearchUIValues, SearchUIActions>>({
+  path: ['enterprise_search', 'app_search', 'search_ui_logic'],
+  actions: () => ({
+    initializeData: () => true,
+    dataInitialized: (initialFieldValues) => initialFieldValues,
+    titleFieldChanged: (titleField) => ({ titleField }),
+    URLFieldChanged: (urlField) => ({ urlField }),
+    facetFieldsChanged: (facetFields) => ({ facetFields }),
+    sortFieldsChanged: (sortFields) => ({ sortFields }),
+    activeFieldChanged: (activeField) => ({ activeField }),
```

Comment:
[optional, naming comment] `xFieldChanged` reads like a boolean/flag to me - I didn't realize they were onChange fns initially. Maybe consider `onXFieldChange` for an action name instead?

Reply:
39ab2d8290ab7111189936084bb7100081d91d49